packages/core/src/external/carequality/ihe-gateway-v2/outbound/xca/process/dq-response.ts

```diff
@@ -118,7 +118,7 @@ function parseDocumentReference(
   }
 
   const documentReference: DocumentReference = {
-    homeCommunityId: stripUrnPrefix(extrinsicObject._home),
+    homeCommunityId: outboundRequest.gateway.homeCommunityId,
```

Comment:
fix for redox.

Reply:
with this, theCommunityid of response will be same as request and will always have a DR sent for it

Reply:
What about the hypothesis in which a DQ responding GW sends a diff home community ID for DR?

If this is specific for a certain GW, we could have a map to have custom behaviors for each GW, something like:

```ts
function defaultGetHomeCommunityIdForDr(
  request: OutboundDocumentQueryReq,
  response: OutboundDocumentQueryResp
): string {
  return response.gateway.homeCommunityId;
}

function getRequestHomeCommunityId(
  request: OutboundDocumentQueryReq,
  response: OutboundDocumentQueryResp
): string {
  return request.gateway.homeCommunityId;
}

const customGetHomeCommunityIdForDr: Record<string, typeof getHomeCommunityIdForDr> = {
  "whatever-oid": getRequestHomeCommunityId,
};

function getHomeCommunityIdForDr(
  request: OutboundDocumentQueryReq,
  response: OutboundDocumentQueryResp
): string {
  const customGet = customGetHomeCommunityIdForDr[request.gateway.homeCommunityId];
  if (customGet) return customGet(request, response);
  return defaultGetHomeCommunityIdForDr(request, response);
}

...

const documentReference: DocumentReference = {
  homeCommunityId: getHomeCommunityIdForDr(outboundRequest, outboundResponse),
  repositoryUniqueId,
  docUniqueId,
  ...

```

Reply:
is that a scenario that we are aware of has ever happened? The only thing I have seen in my testing is endpoints throwing errors when presented with a different homeCommunityId in the gateway and in the Document Reference. 

Also the spec makes it clear that the home Community Id is globally unique, and so if the response has a different id than the request, to me, its clear that the response has something wrong. 

https://profiles.ihe.net/ITI/TF/Volume2/ITI-38.html#3.38.4.1.2.1

Reply:
Ok so I dug deeper and your right. Surescripts is an example of a gateway where they return a different homeCommunityID on the response than the request, and where you need to use that homeCommunityID in the DR Request in order to get the document. 

The way I found this is that I was confused why we have 0 DR responses from Surescripts in the DR retrieval responses. Initially I thought this was because surescripts responds with MTOM and mirth cant handle MTOM. But then it turns out that most DR endpoints actually respond with MTOM, and part of the Mirth interop plugin under the hood is that it parses MTOM. So then I was like wtf. 

And the reason again is this: https://github.com/metriport/metriport/blob/f4b5f7a179bd0068324c1320389c5f84ba98829c/packages/api/src/external/carequality/document/create-outbound-document-retrieval-req.ts#L33

Reply:
So basically we need to do two things here. 

1. Modify the logic in create-outbound-document-retrieval-req.ts to not filter out if they have different ids. 
2. Have the gateway specific logic @leite08 brought up that uses some flag to distinguish the cases where they dont want a different homeCommunityId in the response (redox is the only one we know of right now)

Reply:
What this means also is if we modify create-outbound-document-retrieval-req.ts, we should start getting surescripts data